model/PermissionsStrategyAbstract.php

```diff
@@ -98,6 +98,9 @@ private function is_assoc(array $array): bool
      */
     public function getDeltaPermissions(array $currentPrivileges, array $privilegesToSet): array
     {
+        $add = [];
+        $remove = [];
+
```

Comment:
Unless we re-add this, we get an error and no permission is changed in case we are only adding or only removing ACLs

`[2023-03-02 13:59:59] UDEMO.WARNING: php error(8): compact(): Undefined variable: remove {"0":"PHPERROR","file":"/var/www/html/taoDacSimple/model/PermissionsStrategyAbstract.php","line":129,"trace":{"trace":[{"file":"/var/www/html/taoDacSimple/model/PermissionsStrategyAbstract.php","line":129,"function":"compact"},{"file":"/var/www/html/taoDacSimple/model/SavePermissionsStrategy.php","line":30,"function":"getDeltaPermissions","class":"oat\\taoDacSimple\\model\\PermissionsStrategyAbstract","type":"->"},{"file":"/var/www/html/taoDacSimple/model/PermissionsService.php","line":71,"function":"normalizeRequest","class":"oat\\taoDacSimple\\model\\SavePermissionsStrategy","type":"->"},{"file":"/var/www/html/taoDacSimple/model/PermissionsService.php","line":93,"function":"saveResourcePermissions","class":"oat\\taoDacSimple\\model\\PermissionsService","type":"->"},{"file":"/var/www/html/taoDacSimple/model/tasks/ChangePermissionsTask.php","line":91,"function":"savePermissions","class":"oat\\taoDacSimple\\model\\PermissionsService","type":"->"},{"function":"__invoke","class":"oat\\taoDacSimple\\model\\tasks\\ChangePermissionsTask","type":"->"},{"file":"/var/www/html/tao/models/classes/taskQueue/Task/CallbackTask.php","line":39,"function":"call_user_func"},{"file":"/var/www/html/tao/models/classes/taskQueue/Worker/AbstractWorker.php","line":103,"function":"__invoke","class":"oat\\tao\\model\\taskQueue\\Task\\CallbackTask","type":"->"},{"file":"/var/www/html/taoTaskQueue/model/LongRunningWorker.php","line":109,"function":"processTask","class":"oat\\tao\\model\\taskQueue\\Worker\\AbstractWorker","type":"->"},{"file":"/var/www/html/taoTaskQueue/scripts/tools/RunWorker.php","line":96,"function":"run","class":"oat\\taoTaskQueue\\model\\LongRunningWorker","type":"->"},{"function":"__invoke","class":"oat\\taoTaskQueue\\scripts\\tools\\RunWorker","type":"->"},{"file":"/var/www/html/tao/models/classes/routing/CliController.php","line":59,"function":"call_user_func"},{"file":"/var/www/html/tao/models/classes/mvc/Bootstrap.php","line":231,"function":"runAction","class":"oat\\tao\\model\\routing\\CliController","type":"->"},{"file":"/var/www/html/tao/models/classes/mvc/Bootstrap.php","line":247,"function":"dispatchCli","class":"oat\\tao\\model\\mvc\\Bootstrap","type":"->"},{"file":"/var/www/html/index.php","line":27,"function":"dispatch","class":"oat\\tao\\model\\mvc\\Bootstrap","type":"->"}]}} {"uid":"c134fcd26165fe7431fb29ad"}`